docs/content/tutorial/saving-your-data.mdx

```diff
@@ -3,33 +3,45 @@ title: "Tutorial, part six: Saving your data | Dagster Docs"
 description: Learn how to use I/O managers to save your data.
 ---
 
-# Tutorial, part six: Saving your data
+# Tutorial, part six: Using Dagster to save and load your data
 
-You've written a data pipeline using assets that runs every hour. Currently, your assets are stored in a temporary folder that is deleted after closing the `dagster` process. **Note**: If you set up Dagster without the tutorial, you may have set a `DAGSTER_HOME` environment variable. If so, then your assets will be stored in that directory.
-
-You likely want to store your assets in places such as tables in your database or files in cloud storage. To do this, Dagster has I/O managers that enable reading and writing data to storage systems.
+You've written a data pipeline using assets that runs every hour. Currently, you are manually storing and loading your data using the `json` and `pandas` libraries at the beginning and end of each asset. Dagster provides I/O managers that will handle reading and writing data to storage for you. I/O managers allow you to reduce boilerplate reading and writing code and easily change where your data is stored.
 
 By the end of this section, you will:
 
-- Store your files in a file system
+- Store your files in a file system using an I/O manager
 - Store your data frames as tables in a database
 - Learn about I/O managers
+- Understand when to use I/O managers
 
 ---
 
 ## What are I/O managers?
 
 I/O stands for **input** and **output.** I/O managers are Dagster objects that control how Dagster reads and writes data to specific external services, such as Snowflake or Amazon Web Services (AWS) S3.
 
-They manage **input** by reading an asset from where it’s stored and loading it into memory to be used by a dependent asset. For example, `most_frequent_words` needs `topstories`. Therefore, the I/O manager that `topstories` uses will load the data into memory so the `most_frequent_words` asset can use it.
+They manage **input** by reading an asset from where it’s stored and loading it into memory to be used by a dependent asset. For example, `most_frequent_words` needs `topstories`. Therefore, you could use an I/O manager to load the `topstories` data into memory so that the `most_frequent_words` asset can use it.
 
-I/O managers control **output** by writing the asset's to the location configured. By the end of this section, you’ll configure an I/O manager to write the `hackernews_topstories` DataFrame into a database as a table.
+I/O managers control **output** by writing the asset's to the location configured. By the end of this section, you’ll configure an I/O manager to write the `topstories` DataFrame into a database as a table.
+
+---
+
+## When should you use I/O managers?
```

Comment:
this section is still WIP. I've iterated on it a couple times, but am still not very happy with it. Maybe it would be better at the end of the section when the user is a bit more familiar with I/O managers? Basically I just want the user to feel empowered to decide when to use I/O managers

Reply:
I agree that this would work better at the end of the section.

Worth looking at some of the content here to maybe copy/paste:
- https://docs.dagster.io/guides/dagster/non-argument-deps#when-to-use-assets-without-arguments-and-return-values
- https://docs.dagster.io/tutorial/managing-your-own-io#tutorial-part-7-managing-your-own-io

Reply:
did a bit of work on this section and incorporated things from those two docs